server/src/main/java/org/elasticsearch/common/time/DateFormatters.java

```diff
@@ -1461,6 +1468,86 @@ private static DateFormatter forPattern(String input, Locale locale) {
         }
     }
 
+    static class MergedDateFormatter implements DateFormatter {
+
+        private final String pattern;
+        private final List<DateFormatter> formatters;
+        private final List<DateMathParser> dateMathParsers;
+
+        MergedDateFormatter(String pattern, List<DateFormatter> formatters) {
+            assert formatters.size() > 0;
```

Comment:
the array was switched to a list while copying, I assume that was intentional?

Reply:
Yes that was intentional